ui/perfherder/ReplicatesGraph.jsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular/index.es2015';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCog } from '@fortawesome/free-solid-svg-icons';
+
+import perf from '../js/perf';
+import { errorMessageClass } from '../helpers/constants';
+import ErrorBoundary from '../shared/ErrorBoundary';
+import Graph from '../shared/Graph';
+import PerfSeriesModel from '../models/perfSeries';
+import { getData } from '../helpers/http';
+import { createApiUrl, perfSummaryEndpoint } from '../helpers/url';
+
+// TODO remove $stateParams after switching to react router
+export default class ReplicatesGraph extends React.Component {
+  // TODO: sync parent with children IRT dataLoading
+  constructor(props) {
+    super(props);
+
+    this.state = { dataLoading: true };
+  }
+
+  async componentDidMount() {
+    this.syncGraphData().then(() => {
+      window.document.title = `${this.state.platform}: ${this.state.testName}`;
+      this.setState({ dataLoading: false });
+    });
+  }
+
+  syncGraphData = async () => {
+    const { title } = this.props;
+
+    const { replicateData, drawingData } = await this.fetchReplicateGraphData();
+    const { replicateDataError } = replicateData;
+    const numRuns = drawingData ? drawingData.numRuns : 0;
+
+    return this.setState({
+      drawingData,
+      graphSpecs: this.getGraphSpecs(title, numRuns, replicateDataError),
+    });
+  };
+
+  getGraphSpecs = (title, numRuns, error = false) =>
+    !error
+      ? {
+          target: '',
+          data: [],
+          width: 1000,
+          height: 275,
+          chart_type: 'bar',
+          x_accessor: 'replicate',
+          y_accessor: 'value',
+          legend: '',
+          title: `${title} replicates over ${numRuns} run${
+            numRuns > 1 ? 's' : ''
+          }`,
+        }
+      : {
+          target: '',
+          title: `${title} replicates`,
+          chart_type: 'missing-data',
+          missing_text: 'No Data Found',
+          width: 1000,
+          height: 275,
+        };
+
+  fetchReplicateGraphData = async () => {
+    const { projectName, revision, subtestSignature } = this.props;
+    const replicateData = {};
+
+    const perfDatumResponse = await getData(
+      createApiUrl(perfSummaryEndpoint, {
+        revision,
+        repository: projectName,
+        signature: subtestSignature,
+      }),
+    );
+    const perfDatum = perfDatumResponse.data[0];
+
+    if (perfDatum.values.length === 0) {
+      replicateData.replicateDataError = true;
+      return { replicateData };
+    }
+    const numRuns = perfDatum.values.length; // get it from values.length
+    const replicatePromises = perfDatum.job_ids.map(job_id =>
+      PerfSeriesModel.getReplicateData({ job_id }),
+    );
+
+    return Promise.all(replicatePromises).then(
```

Comment:
I don't see any error handling here, which i think would be preferable to letting the error boundary catch the result of there not being any data. You can do this in a follow up pr since it would require refactoring the `PerfSeriesModel `to use the `getData` wrapper the same way `PushModel.getList` now does (this is something we're making incremental changes to - more details [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1473777)).

Reply:
> 
> 
> I don't see any error handling here, which i think would be preferable to letting the error boundary catch the result of there not being any data. You can do this in a follow up pr since it would require refactoring the `PerfSeriesModel `to use the `getData` wrapper the same way `PushModel.getList` now does (this is something we're making incremental changes to - more details [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1473777)).

Sure thing. I'm filing this new bug and CC you also.